### PR TITLE
friendlier errors?

### LINKED ISSUE #38: Clearer error message for incorrect encryption key
Currently a dropwizard app will fail to start with the following error if the `encrypted-config-value.key` is incorrect:

```
! javax.crypto.AEADBadTagException: Tag mismatch!
! at com.sun.crypto.provider.GaloisCounterMode.decryptFinal(GaloisCounterMode.java:524) ~[sunjce_provider.jar:1.8.0_71]
! at com.sun.crypto.provider.CipherCore.finalNoPadding(CipherCore.java:1023) ~[sunjce_provider.jar:1.8.0_71]
! at com.sun.crypto.provider.CipherCore.doFinal(CipherCore.java:960) ~[sunjce_provider.jar:1.8.0_71]
! at com.sun.crypto.provider.CipherCore.doFinal(CipherCore.java:824) ~[sunjce_provider.jar:1.8.0_71]
! at com.sun.crypto.provider.AESCipher.engineDoFinal(AESCipher.java:436) ~[sunjce_provider.jar:1.8.0_71]
! at javax.crypto.Cipher.doFinal(Cipher.java:2223) ~[na:1.8.0_71]
! at com.palantir.config.crypto.algorithm.AesAlgorithm$2.get(AesAlgorithm.java:100) ~[encrypted-config-value-0.1.5.jar:na]
! at com.palantir.config.crypto.algorithm.AesAlgorithm$2.get(AesAlgorithm.java:88) ~[encrypted-config-value-0.1.5.jar:na]
! at com.palantir.config.crypto.util.Suppliers.silently(Suppliers.java:38) ~[encrypted-config-value-0.1.5.jar:na]
! ... 15 common frames omitted
! Causing: java.lang.RuntimeException: the padding was invalid
! at com.palantir.config.crypto.util.Suppliers.silently(Suppliers.java:42) ~[encrypted-config-value-0.1.5.jar:na]
! at com.palantir.config.crypto.algorithm.AesAlgorithm.getDecryptedString(AesAlgorithm.java:88) ~[encrypted-config-value-0.1.5.jar:na]
! at com.palantir.config.crypto.EncryptedValue.getDecryptedValue(EncryptedValue.java:40) ~[encrypted-config-value-0.1.5.jar:na]
! at com.palantir.config.crypto.EncryptedValue.getDecryptedValue(EncryptedValue.java:53) ~[encrypted-config-value-0.1.5.jar:na]
! at com.palantir.config.crypto.DecryptingVariableSubstitutor$DecryptingStringLookup.lookup(DecryptingVariableSubstitutor.java:31) ~[encrypted-config-value-bundle-0.1.5.jar:na]
! at org.apache.commons.lang3.text.StrSubstitutor.resolveVariable(StrSubstitutor.java:922) ~[commons-lang3-3.4.jar:3.4]
! at org.apache.commons.lang3.text.StrSubstitutor.substitute(StrSubstitutor.java:843) ~[commons-lang3-3.4.jar:3.4]
```

The true cause is unintuitive, and we should be able to provide a more helpful error. Ideally something like `The encrypted value for key 'auth.password' could not be decrypted`

@sarahfbeller for SA

### PR DESCRIPTION
**Pull request by bavardage**
fixes #38

### CODE REVIEW

**Comment by bavardage** [comment]
...not convinced of the approach here...

File: encrypted-config-value-bundle/src/test/resources/bad.key [hunk 1]
```diff
@@ -0,0 +1 @@
+AES:OMG
```
Comment (bmoylan) [inline encrypted-config-value-bundle/src/test/resources/bad.key:1]: Is there a difference between a real 256 bit AES key that can't decrypt the secret vs. invalid AES?

**Comment by bmoylan** [comment]
@bavardage who's a good person to CR this and get it in?

**Comment by bavardage** [comment]
OBE #43

**Pull request closed without merging**